src/alist.c

```diff
@@ -962,3 +962,64 @@ void alist_polef(
 
     dram_store_u16(hle, (uint16_t*)(dst - 4), address, 4);
 }
+
+void alist_iirf(
+        struct hle_t* hle,
+        bool init,
+        uint16_t dmemo,
+        uint16_t dmemi,
+        uint16_t count,
+        int16_t* table,
+        uint32_t address)
+{
+    int16_t *dst = (int16_t*)(hle->alist_buffer + dmemo);
+    int32_t i, prev;
+	int16_t frame[8];
+    int16_t ibuf[4];
+    uint16_t index = 7;
+
+
+    count = align(count, 16);
+
+    #define vmulf(a, b)     (((a)*(b)+0x4000)>>15)
+
+    if(init)
+    {
+        for(i = 0; i < 8; ++i)
+            frame[i] = 0;
+        ibuf[1] = 0;
+        ibuf[2] = 0;
+    }
+    else
+    {
+        frame[6] = *dram_u16(hle, address + 4);
+        frame[7] = *dram_u16(hle, address + 6);
+        ibuf[1] = (int16_t)*dram_u16(hle, address + 8);
+        ibuf[2] = (int16_t)*dram_u16(hle, address + 10);
+    }
+
+    prev = vmulf(table[9], frame[6]) * 2;
+    do
+    {
+        for(i = 0; i < 8; ++i)
+        {
+            int32_t accu;
+			ibuf[index&3] = *alist_s16(hle, dmemi);
+
+            accu = prev + vmulf(table[0], ibuf[index&3]) + vmulf(table[1], ibuf[(index-1)&3]) + vmulf(table[0], ibuf[(index-2)&3]);
+            accu += vmulf(table[8], frame[index]) * 2;
+            prev = vmulf(table[9], frame[index]) * 2;
+            dst[i^S] = frame[i] = accu;
+
+            index=(index+1)&7;
+            dmemi += 2;
+        }
+        dst += 8;
+        count -= 0x10;
+    } while (count > 0);
+
+    dram_store_u16(hle, &frame[6], address + 4, 4);
+    dram_store_u16(hle, (int16_t*)&ibuf[(index-2)&3], address+8, 2);
+    dram_store_u16(hle, (int16_t*)&ibuf[(index-1)&3], address+10, 2);
```

Comment:
for the 3 dram_store_u16 you should cast to (uint16_t*) the second parameter to avoid signedness warnings in gcc.